code/modules/station_goals/bsa.dm

```diff
@@ -213,8 +213,13 @@
 
 	point.Beam(get_target_turf(), icon_state = "bsa_beam", time = 50, maxdistance = world.maxx, beam_type = /obj/effect/ebeam/deadly) //ZZZAP
 	playsound(src, 'sound/machines/bsa_fire.ogg', 100, 1)
+	if(istype(target,/obj/item/gps))
+		var/obj/item/gps/G = target
+		message_admins("[key_name_admin(user)] has launched an artillery strike at GPS named [G.gpstag].")
+
+	else
+		message_admins("[key_name_admin(user)] has launched an artillery strike.")//Admin BSA firing, just targets a room, which the explosion says
```

Comment:
Maybe a stupid question but would it be worth having a an ADMINJMP to the target's coordinates if theres no GPS?

Reply:
explosion has the jump attached to it